pyqtgraph/tests/image_testing.py

```diff
@@ -0,0 +1,520 @@
+        self.failBtn = QtGui.QPushButton('Fail')
+        self.layout.addWidget(self.passBtn, 2, 0)
+        self.layout.addWidget(self.failBtn, 2, 1)
+
+        self.views = (self.view.addViewBox(row=0, col=0),
+                      self.view.addViewBox(row=0, col=1),
+                      self.view.addViewBox(row=0, col=2))
+        labelText = ['test output', 'standard', 'diff']
+        for i, v in enumerate(self.views):
+            v.setAspectLocked(1)
+            v.invertY()
+            v.image = ImageItem()
+            v.image.setAutoDownsample(True)
+            v.addItem(v.image)
+            v.label = TextItem(labelText[i])
+            v.setBackgroundColor(0.5)
+
+        self.views[1].setXLink(self.views[0])
+        self.views[1].setYLink(self.views[0])
+        self.views[2].setXLink(self.views[0])
+        self.views[2].setYLink(self.views[0])
+
+    def test(self, im1, im2, message):
+        """Ask the user to decide whether an image test passes or fails.
+        
+        This method displays the test image, reference image, and the difference
+        between the two. It then blocks until the user selects the test output
+        by clicking a pass/fail button or typing p/f. If the user fails the test,
+        then an exception is raised.
+        """
+        self.show()
+        if im2 is None:
+            message += '\nImage1: %s %s   Image2: [no standard]' % (im1.shape, im1.dtype)
+            im2 = np.zeros((1, 1, 3), dtype=np.ubyte)
+        else:
+            message += '\nImage1: %s %s   Image2: %s %s' % (im1.shape, im1.dtype, im2.shape, im2.dtype)
+        self.label.setText(message)
+        
+        self.views[0].image.setImage(im1.transpose(1, 0, 2))
+        self.views[1].image.setImage(im2.transpose(1, 0, 2))
+        diff = makeDiffImage(im1, im2).transpose(1, 0, 2)
+
+        self.views[2].image.setImage(diff)
+        self.views[0].autoRange()
+
+        while True:
+            QtGui.QApplication.processEvents()
+            lastKey = self.lastKey
+            
+            self.lastKey = None
+            if lastKey in ('f', 'esc') or not self.isVisible():
+                raise Exception("User rejected test result.")
+            elif lastKey == 'p':
+                break
+            time.sleep(0.03)
+
+        for v in self.views:
+            v.image.setImage(np.zeros((1, 1, 3), dtype=np.ubyte))
+
+    def keyPressEvent(self, event):
+        if event.key() == QtCore.Qt.Key_Escape:
+            self.lastKey = 'esc'
+        else:
+            self.lastKey = str(event.text()).lower()
+
+
+def getTestDataRepo():
+    """Return the path to a git repository with the required commit checked
+    out.
+
+    If the repository does not exist, then it is cloned from
+    https://github.com/vispy/test-data. If the repository already exists
+    then the required commit is checked out.
+    """
+    global testDataTag
+
+    dataPath = os.path.expanduser('~/.pyqtgraph/test-data')
```

Comment:
Might want to switch this to `os.path.join(os.path.expanduser('~'), '.pyqtgraph', 'test-data')` to keep it win compatible